src/System.Management.Automation/singleshell/config/MshSnapinInfo.cs

```diff
@@ -1010,13 +1014,20 @@ internal static Collection<PSSnapInInfo> ReadEnginePSSnapIns()
 
             // create default mshsnapininfo objects..
             Collection<PSSnapInInfo> engineMshSnapins = new Collection<PSSnapInInfo>();
+            string assemblyVersionString = assemblyVersion.ToString();
 
             for (int item = 0; item < DefaultMshSnapins.Count; item++)
             {
                 DefaultPSSnapInInformation defaultMshSnapinInfo = DefaultMshSnapins[item];
 
-                string strongName = string.Format(CultureInfo.InvariantCulture, "{0}, Version={1}, Culture={2}, PublicKeyToken={3}, ProcessorArchitecture={4}",
-                    defaultMshSnapinInfo.AssemblyName, assemblyVersion.ToString(), culture, publicKeyToken, architecture);
+                string strongName = string.Format(
+                    CultureInfo.InvariantCulture,
+                    "{0}, Version={1}, Culture={2}, PublicKeyToken={3}, ProcessorArchitecture={4}",
```

Comment:
.Net Core does not return ProcessorArchitecture in Assembly.FullName (aka strong name).
Maybe remove this from our code too?

Reply:
It might be fine to remove, I don't know for sure. I wouldn't do it as it doesn't really bring any benefit.

Reply:
One code path leads too InitializeTypeCatalog() - there is no ProcessorArchitecture :-(
```c#
        private Dictionary<string, string> InitializeTypeCatalog()
        {
            Dictionary<string, string> typeCatalog = new Dictionary<string, string>(6217, StringComparer.OrdinalIgnoreCase);
            typeCatalog["Markdig.IMarkdownExtension"] = "Markdig.Signed, Version=0.17.1.0, Culture=neutral, PublicKeyToken=870da25a133885f8";
            typeCatalog["Markdig.Markdown"] = "Markdig.Signed, Version=0.17.1.0, Culture=neutral, PublicKeyToken=870da25a133885f8";
```

Reply:
I know. You need to review all usage of the `PSSnapInInfo.AssemblyName`.